core/app/models/spree/calculator/weighted_rate.rb

```diff
@@ -0,0 +1,19 @@
+require_dependency 'spree/calculator'
+
+module Spree
+  class Calculator::WeightedRate < Calculator
+    preference :amount, :decimal, default: 0
+    preference :currency, :string, default: -> { Spree::Config[:currency] }
+
+    def compute_line_item(line_item)
+      if line_item && preferred_currency.casecmp(line_item.currency).zero?
```

Comment:
I'm not sure this is worth fixing, but we check the currency here at the line level but we don't check it again in WeightedAdjustmentsHandler.  That's fine in real life since we know that behind the scenes the line item delegates the currency to the order, but it seems wrong somehow to have to know that?